src/plugins/InlineSpoilers/Plugin.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Mark,
+  markInputRule,
+  markPasteRule,
+  mergeAttributes,
+} from "@tiptap/core";
+
+import { PluginKey } from "@tiptap/pm/state";
+import { toggleAttributeOnClick } from "../utils";
+
+export interface InlineSpoilersOptions {
+  visible?: boolean;
+  focusable?: boolean;
+}
+
+export const InlineSpoilersPluginKey = new PluginKey("InlineSpoilersPlugin");
+
+export const PLUGIN_NAME = "inlineSpoilers";
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    [PLUGIN_NAME]: {
+      setInlineSpoilers: (attributes?: { visible: boolean }) => ReturnType;
+      toggleInlineSpoilers: (attributes?: { visible: boolean }) => ReturnType;
+      unsetInlineSpoilers: () => ReturnType;
+    };
+  }
+}
+
+export const inputRegex = /(?:^|\s)((?:\|\|)((?:[^\|]+))(?:\|\|))$/;
+export const pasteRegex = /(?:^|\s)((?:\|\|)((?:[^\|]+))(?:\|\|))/g;
+
+const toggleSpoilersOnKeydown = (event: KeyboardEvent) => {
+  console.log("in keydown event");
+  if (
+    event.key !== "R" ||
+    event.ctrlKey ||
+    event.metaKey ||
+    !event.altKey ||
+    !event.shiftKey
+  ) {
+    console.log("no key match");
+    return;
+  }
+  if (document.activeElement?.getAttribute("data-type") !== PLUGIN_NAME) {
+    console.log("activeElement", document.activeElement);
+    return;
+  }
+  const spoilersElement = document.activeElement;
+  const currentValue = spoilersElement.getAttribute("data-visible");
+  if (!currentValue) {
+    console.log("element attribute has no currentValue");
+    return;
+  }
+  const newValue = currentValue === "false" ? "true" : "false";
+  console.log(`toggling data-visible from ${currentValue} to ${newValue}`);
+  spoilersElement.setAttribute("data-visible", newValue);
+};
+
+export const InlineSpoilersPlugin = Mark.create<InlineSpoilersOptions>({
+  name: PLUGIN_NAME,
+  priority: 1001,
+
+  addAttributes() {
+    return {
+      visible: {
+        default: false,
+        parseHTML: (element) => element.getAttribute("data-visible"),
+        renderHTML: (attributes) => {
+          return {
+            "data-visible": attributes.visible,
+          };
+        },
+      },
+    };
+  },
+
+  addOptions() {
+    return {
```

Comment:
I think this is fine for now, but I'd like to figure out a better way with time.

TBH it feels like the better way would be to find out how to tell all plugins whether they're being rendered as part of an editable editor or not, so they can react accordingly. It seems like this would be broadly useful.

Reply:
You have access to `this.editor` (and thus `this.editor.editable`) in many of the plugin fields, but not `renderHTML()`, `parseHTML()`, `addOptions()` and `addAttributes()`. Presumably there is a reason you specifically can't access it there, but my crime-committing skills are not yet up to figuring out how to get around it.

Reply:
That's interesting. Yeah, feel free to leave it for now.